### DISCUSSION TITLE
InsertMany

### DISCUSSION THREAD

**Discussion by psicomante** [maxnowack/signaldb#919] [General]
Hello max, great library!

I would wonder why it has not been implemented an InsertMany method (like bulkInsert in RxDB). Maybe because of Mingo?

Or could it be done in other ways?

Thanks in advance.
Rob

**Comment by maxnowack** [comment]
Hey @psicomante, thanks!
There wasn't a need for an `insertMany` function as looping over the items and inserting them should be around the same speed. It just needs to be wrapped in a `batch` method:

```js
collection.batch(() => {
  items.forEach((item) => {
    collection.insert(item)
  })
})
```

But I think a `insertMany` function would make sense in terms of syntactic sugar, so feel free to create a PR 🙂

**Reply by psicomante** [reply]
Thanks Max for your reply! I saw that the batch has been added just two days ago, amazing, I was using an older version. This library is growing fast!

I just cloned the repo, I will definitely take confidence with the code to submit a PR :)
